core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -355,4 +329,38 @@ private static String featureName(final URL url) {
         // fallback
         return Math.abs(url.hashCode()) + Features.EXTENSION;
     }
+
+    private static boolean isCukeSpaceTest(final TestClass testClass, final ClassLoader loader) {
+        Class<? extends Annotation> runWithType = null;
+        try {
+            runWithType = (Class<? extends Annotation>) loader.loadClass("org.junit.runner.RunWith");
+
+        } catch (final ClassNotFoundException error) {
+            // no-op
+        } catch (final NoClassDefFoundError e) {
+            // no-op
+        }
+
+        if (runWithType == null) {
+            return false;
+        }
+        final Annotation runWith = testClass.getAnnotation(runWithType);
+        if (runWith == null) {
+            return false;
+        }
+
+        try {
+            final Class<?> runner = Class.class.cast(runWithType.getMethod("value").invoke(runWith));
+            if ((!ArquillianCucumber.class.equals(runner) && !CukeSpace.class.equals(runner))) {

```

Comment:
My company is wrapping Cucumber to provide some customizations for our product.
This means that using an extension with ArquillianCucumber I don't want ArquillianCucumber to enrich my testable archive but I want my extension instead to do that.
Following your approach both the base  ArquillianCucumber and the extension enriche the testable archives and the last one will be the winner.
This is how I faced the problem we are fixing here :)